kolibri/utils/cli.py

```diff
@@ -167,13 +182,118 @@ def _first_run():
 
     logger.info("Automatically enabling applications.")
 
+    # Finally collect static assets and run migrations again
+    update()
+
+
+def update():
+    """
+    Called whenever a version change in kolibri is detected
+
+    TODO: We should look at version numbers of external plugins, too!
+    """
+    logger.info("Running update routines for new version...")
+    call_command("collectstatic", interactive=False)
+    call_command("collectstatic_js_reverse", interactive=False)
+
+    from kolibri.core.settings import SKIP_AUTO_DATABASE_MIGRATION
+
+    if not SKIP_AUTO_DATABASE_MIGRATION:
+        call_command("migrate", interactive=False, database="default")
+        call_command("migrate", interactive=False, database="ormq")
+
     with open(VERSION_FILE, "w") as f:
         f.write(kolibri.__version__)
 
 
+def start(port=8080, daemon=True):
+    """
+    Start the server on given port.
+
+    :param: port: Port number (default: 8080)
+    :param: daemon: Fork to background process (default: True)
+    """
+
+    if not daemon:
+        logger.info("Running 'kolibri start' in foreground...\n")
+    else:
+        logger.info("Running 'kolibri start' as daemon (system service)\n")
+
+    # TODO: moved from server.start() but not sure where it should ideally be
+    # located. Question is if it should be run every time the server is started
+    # or if it depends on some kind of state change.
+    from kolibri.content.utils.annotation import update_channel_metadata_cache
```

Comment:
I think that is definitely a good caution. Any annotation tasks that are done after the server has already daemonized should be delegated to a task runner.